python/ray/data/preprocessor.py

```diff
@@ -226,59 +254,83 @@ def _transform(self, dataset: Dataset) -> Dataset:
             data_format = "arrow"
+        elif isinstance(data, (dict, np.ndarray)):
+            data_format = "numpy"
         else:
             raise NotImplementedError(
-                "`transform_batch` is currently only implemented for Pandas DataFrames "
-                f"and PyArrow Tables. Got {type(df)}."
+                "`transform_batch` is currently only implemented for Pandas "
+                "DataFrames, PyArrow Tables, Numpy ndarray and dictionary of "
+                f"ndarary. Got {type(data)}."
             )
 
         transform_type = self._determine_transform_to_use(data_format)
 
         if transform_type == "pandas":
             if data_format == "pandas":
-                return self._transform_pandas(df)
-            else:
-                return self._transform_pandas(df.to_pandas())
-        elif transform_type == "arrow":
-            if data_format == "arrow":
-                return self._transform_arrow(df)
+                return self._transform_pandas(data)
             else:
-                return self._transform_arrow(pyarrow.Table.from_pandas(df))
+                return self._transform_pandas(data.to_pandas())
+        elif transform_type == "numpy":
+            if data_format == "numpy":
+                return self._transform_numpy(data)
+            elif data_format == "arrow":
+                if len(data.column_names) == 1:
+                    # If just a single column, return as a single numpy array.
+                    return self._transform_numpy(data[0].to_numpy())
+                else:
+                    output_dict = {}
+                    for col_name in data.column_names:
+                        output_dict[col_name] = data[col_name].to_numpy()
+                return self._transform_numpy(output_dict)
+            elif data_format == "pandas":
+                if len(data.columns) == 1:
+                    # If just a single column, return as a single numpy array.
+                    return self._transform_numpy(data.iloc[:, 0].to_numpy())
+                else:
+                    # Else return as a dict of numpy arrays.
+                    output_dict = {}
+                    for column_name in data:
+                        output_dict[column_name] = data[column_name].to_numpy()
+                    return self._transform_numpy(output_dict)
 
     @DeveloperAPI
     def _transform_pandas(self, df: "pd.DataFrame") -> "pd.DataFrame":
         """Run the transformation on a data batch in a Pandas DataFrame format."""
         raise NotImplementedError()
 
     @DeveloperAPI
-    def _transform_arrow(self, table: "pyarrow.Table") -> "pyarrow.Table":
-        """Run the transformation on a data batch in a PyArrow Table format."""
+    def _transform_numpy(
+        self, np_data: Union[np.ndarray, Dict[str, np.ndarray]]
+    ) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+        """Run the transformation on a data batch in a Numpy array format."""
```

Comment:
Consistent branding.
```suggestion
        """Run the transformation on a data batch in a NumPy ndarray format."""
```